resultshandling/printer/prometheusprinter.go

```diff
@@ -25,74 +29,41 @@ func (prometheusPrinter *PrometheusPrinter) Score(score float32) {
 	fmt.Printf("\n# Overall score out of 100\nkubescape_score %f\n", score*100)
 }
 
-func (printer *PrometheusPrinter) printDetails(details []reporthandling.RuleResponse, frameworkName string, controlName string) error {
-	objs := make(map[string]map[string]map[string]int)
-	for _, ruleResponses := range details {
-		for _, k8sObj := range ruleResponses.AlertObject.K8SApiObjects {
-			kind, ok := k8sObj[`kind`].(string)
-			if (!ok) {
-				return errors.New("Found object with non string kind")
-			}
-			apiVersion,ok := k8sObj[`apiVersion`].(string)
-			if (!ok) {
-				return errors.New("Found object with non string apiVersion")
-			}
-			gvk := fmt.Sprintf("%s/%s",apiVersion,kind)
-			metadata,ok := k8sObj[`metadata`].(map[string]interface{})
-			if (!ok) {
-				return errors.New("Found object with non convertable metadata")
-			}
-			name,ok := metadata[`name`].(string)
-			if (!ok) {
-				return errors.New("Found metadata with non string name")
-			}
-			namespace,ok := metadata[`namespace`].(string)
-			if (!ok) {
-				namespace = ""
-			}
-			if (objs[gvk] == nil) {
-				objs[gvk] = make(map[string]map[string]int)
-			}
-			if (objs[gvk][namespace] == nil) {
-				objs[gvk][namespace] = make(map[string]int)
-			}
-			objs[gvk][namespace][name]++
-		}
+func (printer *PrometheusPrinter) printResources(allResources map[string]workloadinterface.IMetadata, resourcesIDs *reporthandling.ResourcesIDs, frameworkName, controlName string) {
+	printer.printDetails(allResources, resourcesIDs.GetFailedResources(), frameworkName, controlName, "failed")
+	printer.printDetails(allResources, resourcesIDs.GetWarningResources(), frameworkName, controlName, "excluded")
+	if printer.verboseMode {
+		printer.printDetails(allResources, resourcesIDs.GetPassedResources(), frameworkName, controlName, "passed")
 	}
-	for gvk, namespaces := range objs {
-		for namespace, names := range namespaces {
-			for name, value := range names {
-				fmt.Fprintf(printer.writer, "# Failed object from %s control %s\n", frameworkName, controlName)
-				if namespace != "" {
-					fmt.Fprintf(printer.writer, "kubescape_object_failed_count{framework=\"%s\",control=\"%s\",namespace=\"%s\",name=\"%s\",groupVersionKind=\"%s\"} %d\n", frameworkName, controlName, namespace, name, gvk, value)
-				} else {
-					fmt.Fprintf(printer.writer, "kubescape_object_failed_count{framework=\"%s\",control=\"%s\",name=\"%s\",groupVersionKind=\"%s\"} %d\n", frameworkName, controlName, name, gvk, value)
-				}
-			}
+
+}
+func (printer *PrometheusPrinter) printDetails(allResources map[string]workloadinterface.IMetadata, resourcesIDs []string, frameworkName, controlName, status string) {
+	fmt.Fprintf(printer.writer, "# Number of resources found as part of %s control \"%s\"\nkubescape_resources_found_count{framework=\"%s\",control=\"%s\"} %d\n", status, controlName, frameworkName, controlName, len(resourcesIDs))
```

Comment:
kubescape_resources_found_count will now be emitted 3 times with identical labels and 3 (probably) different numbers. It should be just emitted once with the total of all found.
We used to emit _failed_count and _excluded_count and they are not emitted any more.